crates/yttrium/src/chain_abstraction/client.rs

```diff
@@ -365,21 +365,47 @@ impl Client {
 }
 
 #[derive(Debug)]
+#[cfg_attr(feature = "uniffi", derive(uniffi_macros::Record))]
 pub struct RouteUiFields {
-    pub route: Vec<TxnDetails>,
+    pub route_details: Vec<TxnDetails>,
     // pub bridge: TxnDetails,
-    pub initial: TxnDetails,
+    pub initial_details: TxnDetails,
     pub local_total: Amount,
 }
 
-pub type TxnDetails = (Transaction, Eip1559Estimation, TransactionFee);
+#[derive(Debug)]
+#[cfg_attr(feature = "uniffi", derive(uniffi_macros::Record))]
+pub struct TxnDetails {
+    pub transaction: Transaction,
+    pub eip1559: Eip1559EstimationFields,
+    pub transaction_fee: TransactionFee,
+}
 
 #[derive(Debug)]
+#[cfg_attr(feature = "uniffi", derive(uniffi_macros::Record))]
+pub struct Eip1559EstimationFields {
```

Comment:
Can we please move this stringification to `kotlin-ffi` so that the types in `yttrium` remain Rust-idiomatic? I.e. map it in kotlin-ffi's get_ui_fields() into a new set of types

The next version of UniFFI will supporting adding UniFFI to types from external crates like this `Eip1559Estimation` so we can remove that mapping layer